pipeline/pipelines/grpc_generation_pipeline.py

```diff
@@ -14,71 +14,47 @@
 
 """Pipelines that run gRPC codegen"""
 
-from pipeline.pipelines import code_generation_pipeline
+from pipeline.pipelines import code_generation_pipeline as code_gen
 from pipeline.tasks import protoc_tasks, package_tasks, publish_tasks
 
 
-class GrpcClientPipeline(code_generation_pipeline.CodeGenerationPipelineBase):
+class GrpcClientPipeline(code_gen.CodeGenerationPipelineBase):
 
     def __init__(self, **kwargs):
-        super(GrpcClientPipeline, self).__init__(**kwargs)
+        super(GrpcClientPipeline, self).__init__(
+            get_grpc_task_factory(kwargs['language']), **kwargs)
 
-    def get_grpc_codegen_tasks(self, **kwargs):
-        return [protoc_tasks.GrpcPackmanTask('Packman', inject=kwargs)]
 
-    def get_grpc_publish_tasks(self, **kwargs):
-        return []
+class GrpcTaskFactoryBase(object):
```

Comment:
It seems like `TaskFactory` is basically an interface declaring `get_tasks` and `get_validate_kwargs` methods. Is it work formalizing this as an abstract base class, rather than inheriting directly from `object`?